src/modules/tasks/components/TasksList/TasksList.tsx

```diff
@@ -25,32 +30,85 @@ interface DataResponse {
 export default function TasksList() {
   const navigate = useNavigate();
   const [tasks, setTasks] = useState<DataResponse[]>([]);
+  const [nameValue, setNameValue] = useState("");
+  const [arrayOfPages, setArrayOfPages] = useState<number[]>([]);
+  const [taskId, setTaskId] = useState(0);
   const [selectedId, setSelectedId] = useState<number | null>(null);
-
+  const [pageNumber, setPageNumber] = useState(1);
+  const [pageSize, setPageSize] = useState(5);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
   const toggleOptions = (id: number) => {
     if (selectedId === id) {
-      setSelectedId(null); 
+      setSelectedId(null);
     } else {
-      setSelectedId(id); 
+      setSelectedId(id);
     }
   };
 
-  const getAllTasks = async () => {
+  const getAllTasks = async (
+    title: string,
+    pageNumber: number,
+    pageSize: number
+  ) => {
     try {
-      const response = await axiosInstance.get(TASKS_URLS.GET_TASKS_MANAGER);
+      const response = await axiosInstance.get(TASKS_URLS.GET_TASKS_MANAGER, {
+        params: { title: title, pageNumber: pageNumber, pageSize: pageSize },
```

Comment:
It is reccomended for filtration and pagination to use search params to persist the state when the user reload the page or  share it with someone else.